src/Factory/OrderFactory.php

```diff
@@ -75,6 +76,12 @@ private function copyOrderItemsToReorder(OrderInterface $order, OrderInterface $
 
         /** @var OrderItemInterface $orderItem */
         foreach ($orderItems as $orderItem) {
+            /** @var ProductVariantInterface $productVariant */
+            $productVariant = $orderItem->getVariant();
+            if (!($productVariant->isTracked() && $productVariant->isInStock())) {
```

Comment:
IMO this if should look like: `if ($productVariant->isTracked() && !$productVariant->isInStock())`